### ISSUE TITLE
Installing flutter_native_splash 1.0.3 cause version solving failed

### ISSUE THREAD

**Issue by freewebwithme** [jonbhanson/flutter_native_splash#140]
**Describe the bug**
Installing flutter_native_splash 1.0.3 cause `version solving failed`

**Flutter doctor**

[✓] Flutter (Channel stable, 2.0.1, on Linux, locale en_US.UTF-8)
    • Flutter version 2.0.1 at /home/user/flutter
    • Framework revision c5a4b4029c (5 days ago), 2021-03-04 09:47:48 -0800
    • Engine revision 40441def69
    • Dart version 2.12.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /home/user/Android/Sdk
    • Platform android-30, build-tools 30.0.0
    • ANDROID_HOME = /home/user/Android/Sdk
    • Java binary at: /home/user/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✗] Chrome - develop for the web (Cannot find Chrome executable at google-chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Android Studio
    • Android Studio at /home/user/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /home/user/android-studio
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] Connected device (1 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86 • Android 10 (API 29) (emulator)

**Error message when tried flutter pub get**
```
 Because no versions of phoenix_socket match >0.4.11 <0.5.0 and phoenix_socket 0.4.11 depends on web_socket_channel ^1.1.0, phoenix_socket ^0.4.11 requires web_socket_channel ^1.1.0.
    Because graphql >=4.1.0-beta.1 depends on web_socket_channel ^2.0.0 and graphql >=4.0.0-alpha.4 <4.1.0-beta.1 depends on uuid_enhanced ^3.0.2, graphql >=4.0.0-alpha.4 requires web_socket_channel ^2.0.0 or uuid_enhanced ^3.0.2.
    Thus, if phoenix_socket ^0.4.11 and graphql >=4.0.0-alpha.4 then uuid_enhanced ^3.0.2.
    And because no versions of uuid_enhanced match >3.0.2 <4.0.0, if phoenix_socket ^0.4.11 and graphql >=4.0.0-alpha.4 then uuid_enhanced 3.0.2.
    And because uuid_enhanced 3.0.2 depends on crypto ^2.0.0 and archive >=3.0.0 depends on crypto ^3.0.0, one of phoenix_socket ^0.4.11 or graphql >=4.0.0-alpha.4 or archive >=3.0.0 must be false.
(1) So, because flutter_native_splash >=1.0.1 depends on image ^3.0.1 which depends on archive ^3.0.0, one of phoenix_socket ^0.4.11 or graphql >=4.0.0-alpha.4 or flutter_native_splash >=1.0.1 must be false.
                                                                        
    Because no versions of graphql_flutter match >4.0.1 <4.0.2-beta.1 and graphql_flutter >=4.0.2-beta.1 <4.0.3-alpha.1 depends on graphql ^4.0.2-beta.1, graphql_flutter >4.0.1 <4.0.3-alpha.1 requires graphql ^4.0.2-beta.1.
    And because graphql_flutter >=4.0.3-alpha.1 <4.1.0-beta.1 depends on graphql ^4.0.3-alpha.1, graphql_flutter >4.0.1 <4.1.0-beta.1 requires graphql ^4.0.2-beta.1.
    And because graphql_flutter >=4.1.0-beta.1 <5.0.0-nullsafety.1 depends on graphql ^4.1.0-beta.1 and graphql_flutter 4.0.1 depends on graphql ^4.0.1, graphql_flutter >=4.0.1 <5.0.0-nullsafety.1 requires graphql ^4.0.1.
    And because one of phoenix_socket ^0.4.11 or graphql >=4.0.0-alpha.4 or flutter_native_splash >=1.0.1 must be false (1), one of phoenix_socket ^0.4.11 or flutter_native_splash >=1.0.1 or graphql_flutter >=4.0.1 <5.0.0-nullsafety.1 must be false.
    And because myapp depends on graphql_flutter ^4.0.1, phoenix_socket ^0.4.11 is incompatible with flutter_native_splash >=1.0.1.
    So, because myapp depends on both phoenix_socket ^0.4.11 and flutter_native_splash ^1.0.3, version solving failed.
Running "flutter pub get" in myapp...                        
pub get failed (1;     So, because myapp depends on both phoenix_socket ^0.4.11 and flutter_native_splash ^1.0.3, version solving failed.)
```

**Comment by jonbhanson** [comment]
It looks like maybe you're not using null safety? If you are using Flutter 1.x (no null safety), you must use the [0.x version](https://pub.dev/packages/flutter_native_splash/versions) of this package.

**Comment by freewebwithme** [comment]
Am I using Flutter 2.0.1? as shown
 `[✓] Flutter (Channel stable, 2.0.1, on Linux, locale en_US.UTF-8)
• Flutter version 2.0.1 at /home/user/flutter`

**Comment by jonbhanson** [comment]
It looks like the `phoenix_socket` package does not have sound null-safety support.  The repo has an open [issue](https://github.com/braverhealth/phoenix-socket-dart/issues/23) regarding this.  Since `flutter_native_splash` 1.x has null support, the two packages are incompatible.  Read more here: [Migrating to null safety](https://dart.dev/null-safety/migration-guide).

**Comment by matehat** [comment]
phoenix_socket with nullsafety support has been published, as version 0.5.0

**Comment by freewebwithme** [comment]
Thanks! @jonbhanson @matehat
